sound/soc/sof/ipc4-topology.c

```diff
@@ -1308,6 +1308,25 @@ sof_ipc4_update_resource_usage(struct snd_sof_dev *sdev, struct snd_sof_widget *
 	pipeline = pipe_widget->private;
 	pipeline->mem_usage += total;
 
+	/*
+	 * If this is not a Data Processing module instance, add the
+	 * required heap sizes to the sum of all modules instance's
+	 * belonging to same pipeline and find the maximun stack
+	 * requirement of all module instances belonging to the same
+	 * pipeline. Non DP module instances should use pipeline_id as
+	 * their domain_id.
+	 */
+	if (swidget->comp_domain != SOF_COMP_DOMAIN_DP) {
+		pipe_widget->heap_bytes += swidget->heap_bytes;
+		if (pipe_widget->stack_bytes < swidget->stack_bytes)
+			pipe_widget->stack_bytes = swidget->stack_bytes;
+
+		if (swidget->domain_id != 0)
+			dev_warn(sdev->dev, "%s is not DP, but domain_id %u was set in topology",
+				 swidget->widget->name, swidget->domain_id);
```

Comment:
if we've going to override it anyway, do we need a warning at all? Wont it be just noise?

Reply:
This logic is about to change. We need separate domain_id and pipeline_id. So this code is already obsolete, but I don't try to reimplement it yet, since it seems the target is still moving.